wctl/counter.go

```diff
@@ -0,0 +1,50 @@
+package wctl
+
+import (
+	"sync"
+	"sync/atomic"
+	"unsafe"
+)
+
+type Counter struct {
+	n    uint64
+	lock sync.RWMutex
```

Comment:
I feel this is a good time for us to introduce https://github.com/uber-go/atomic. Thoughts?

Reply:
Good idea.